types/lodash/lodash-tests.ts

```diff
@@ -3519,6 +3519,17 @@ fp.now(); // $ExpectType number
 
 // _.negate
 {
+    const f: (...args: number[]) => boolean = anything;
+    // Ideally these would capture all parameters, but it's not easily done
+    // without generic rest parameters, which requires TS 3.0, which these types
+    // don't currently support.
+    // Note: these tests should be tested without `strictFunctionTypes`, as
+    // overload behaviour differs in that case.
+    _.negate(f); // $ExpectType (a1: number) => boolean
+    _(f).negate(); // $ExpectType LoDashImplicitWrapper<(a1: number) => boolean>
+    _.chain(f).negate(); // $ExpectType LoDashExplicitWrapper<(a1: number) => boolean>
+    fp.negate(f); // $ExpectType (a1: number) => boolean
+
```

Comment:
Prior to the overload change made in this PR, these tests would fail when `strictFunctionTypes` was disabled.

The 0 parameter overload would be used instead.

Changing the order of the overloads helps, but it doesn't completely fix the problem.

IIUC, the only way to fix it properly is to add an overload with generic rest parameters:

``` ts
        negate<T extends any[]>(predicate: (...args: T) => any): (...args: T) => boolean;
```

However, this requires TS 3.0, which these types do not currently support.